### DISCUSSION TITLE
How to upgrade single-node cluster?

### DISCUSSION THREAD

**Discussion by sj14** [siderolabs/talos#6554] [Q&A] [answered]
I think this method worked with an earlier version of talos/talosctl but it seems to have stopped working now. I've tried kind of all variants with those flags:

```
➜  ~ talosctl upgrade --image ghcr.io/siderolabs/installer:v1.2.7 --force --stage --wait
◳ watching nodes: [<IP>]
    * <IP>: rpc error: code = Unknown desc = error validating etcd for upgrade: only 1 etcd member found. assuming this is not an HA setup and refusing to upgrade
```

```
➜  ~ talosctl version
Client:
	Tag:         v1.2.6
	SHA:         6ca8b2ac
	Built:
	Go version:  go1.19.2
	OS/Arch:     darwin/amd64
Server:
	NODE:        <IP>
	Tag:         v1.2.6
	SHA:         6ca8b2ac
	Built:
	Go version:  go1.19.2
	OS/Arch:     linux/arm64
	Enabled:     RBAC
```

It's a single-node cluster where HA is not important.

**Comment by steverfrancis** [comment, marked as answer]
You just need the `—préserve` flag.

from https://www.talos.dev/v1.2/talos-guides/upgrading-talos/

“ There is an option to this command: --preserve, which will explicitly tell Talos to keep ephemeral data intact. In most cases, it is correct to let Talos perform its default action of erasing the ephemeral data. However, if you are running a single-node control-plane, you will want to make sure that --preserve=true.”
